Content.Server/GameTicking/Rules/RandomRuleSystem.cs

```diff
@@ -0,0 +1,80 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.StationEvents;
+using Content.Shared.GameTicking.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Content.Shared.Storage;
+using Content.Server.StationEvents.Components;
+using System.Linq;
+
+namespace Content.Server.GameTicking.Rules;
+
+
+/// <summary>
+/// Entities spawned with this component will try to add gamerules from the provided EntitySpawnEntry list.
+/// This can be used to subset gamerules for balance, or to do something else like add it to a prototype if you want ninjas to start with meteors etc.
+/// </summary>
+public sealed class RandomRuleSystem : GameRuleSystem<RandomRuleComponent>
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IChatManager _chatManager = default!;
+    [Dependency] private readonly EventManagerSystem _event = default!;
+
+    protected override void Added(EntityUid uid, RandomRuleComponent component, GameRuleComponent gameRule, GameRuleAddedEvent args)
```

Comment:
this should probably happen on `Started` instead of `Added`

Reply:
See last review for why Added may be better.